### PR TITLE
New ban type: Observer only

### PR DESCRIPTION
**Pull request by Securitymain**
## What this does

Adds the special type of jobban "observer":
- Can't join the game as anything but observer
- Won't be polled for midround recruitment
- Can't become mouse, mommi, hobo or pAI
- They don a dunce cap

Added to the admin player panel
Added cluwne ban to the panel as well, was missing

## Why it's good

You just want to give a shitter a timeout but you want to let them observe the game still? Then this ban is for you. They can still use deadchat and OOC. If they start shitting those up, then a regular ban was the right choice after all.

observer bans

## Changelog

:cl:
 * rscadd: Added "observer" type ban, can only observe. Added alongside cluwne to admin player panel jobban list.

### CODE REVIEW

**Comment by Securitymain** [comment]
There may be loopholes, but I know someone who would be the perfect guinea pig for some live testing

**Comment by MercuriusGH** [comment]
Can admins still shove observerbanned people into mobs?
I think it would be good if there were an option to parole the convicts for busses and whatnot

**Comment by Inorien** [comment]
Only if you can't chat or vote either

**Comment by zakanater19** [comment]
Should be the regular ban type, including for permabans, if it's implemented unless the ban has to do with anything deadchat/OOC/metagaming using ghost.

**Comment by zakanater19** [comment]
> Only if you can't chat or vote either

they can't vote anyway unless they play in the round

**Comment by Inorien** [comment]
its funny that you (alone) upvoted the discord bot notification of this issue

**Comment by Eneocho** [comment]
+1 if the bans page finally gets updated to show all the special ban types properly. It needs a third section that only shows the special bans, such as pax ban, cluwne ban, observer ban, etc.

**Comment by Eneocho** [comment]
Make it so observer banned players have a "wake up" button that spawns them in hell, that one area for the permabanned player mobs.

**Comment by zakanater19** [comment]
> its funny that you (alone) upvoted the discord bot notification of this issue

who. me?

**Comment by Securitymain** [comment]
> Can admins still shove observerbanned people into mobs? I think it would be good if there were an option to parole the convicts for busses and whatnot

yeah, if they ghost out from that they get the regular ghost verbs though, including mommi and hobo. Only the obsghost gets those deleted

**Comment by SonixApache** [comment]
this should be added with the hard exception that ~~you~~ securitymain gets full-banned rather than observer-banned no matter what

**Comment by pedr69** [comment]
what did sonix mean by this?
this should be added but never used?

**Comment by Hinaichigo** [comment]
The dunce cap works with all of the outfits right?
Might I suggest putting the logic for it in
modules/mob/new_player/preferences_setup.dm, with the other observer clothing stuff?

I like this concept as being compelled to watch other people play could be a nice way to rectify  certain behaviors.

**Comment by Securitymain** [comment]
> The dunce cap works with all of the outfits right? Might I suggest putting the logic for it in modules/mob/new_player/preferences_setup.dm, with the other observer clothing stuff?

Unless I missed something you don't have access to a player client/mob in there, which is required for a ban check

**Comment by Securitymain** [comment]
After chewing on it a bit longer I agree. Downside is the function signature changing, as the client needs passing down... whatever

**Comment by angryturnip** [comment]
I think removing the use of OOC would be a good idea to be bundled in with this ban type, personally

**Comment by Securitymain** [comment]
> I think removing the use of OOC would be a good idea to be bundled in with this ban type, personally

OOC bans already exist, so admins can just combine the two.
I get where you're coming from, though

**Comment by angryturnip** [comment]
> OOC bans already exist, so admins can just combine the two. I get where you're coming from, though

I realize it's already a thing, but at the same time pretty much the first thing a disgruntled person that's obs banned would do is ick ock in ooc. I wouldn't even mind if they couldn't use deadchat either, but I don't ultimately care either way on that.

**Comment by Irradiation** [comment]
Cluwnebans are already a thing if you want to worm your way back into the game despite being banned, this is just another step (with a lot of >unforeseen consequences) on the slippery slope that Cluwnebans started.

**Comment by NigglyWiggly** [comment]
This would literally cause nothing but problems.
The point of a ban is to remove someone from the game, temporarily or otherwise. If you make it so they can still come observe everything, even if they can't directly interact with anyone at all after having to apply 5 other bans, they can still just join and ick ock with screenshots and generally hang around when they're not supposed to be doing anything on the server to begin with.

**Comment by wotisjanitor** [comment]
I actually thought this was how bans worked wtf. This should be the default

**Comment by KingofHONK** [comment]
No issues from the headmins on this. I do not see this type of ban being applied often, and we will definately want to use this sparingly.

> this should be added with the hard exception that ~you~ securitymain gets full-banned rather than observer-banned no matter what

Agreed. Someone coding an alternate ban for themselves and for it getting applied to them retroactively is gay for very obvious reasons and wont be happening here.

**Comment by d3athrow** [comment]
This is stupid

**Comment by KingofHONK** [comment]
Honestly just paypal me 20 and ill stream rounds to you live.

**Comment by bathosbathtime** [comment]
I don't fully understand when you would ban someone, but only a little bit, but having the option is fine. I doubt this will be used much if at all. In the past we've been able to soft observer ban people by job banning them from all jobs so there was nothing really stopping this from being done before, though it is nice for it for sure not poll for midrounds, which is the one concern I'd have for the all jobbans method after the dynamic update which is probably the only actual concern for using this method otherwise.
But typically we don't want to encourage people to hang around a community they're banned from. It encourages them to find ways to subvert the game in unexpected ways and frankly doesn't seem healthy to me for their sake even if they're benign.
Even still, I'm not going to veto or even vote against this PR as more tools in the admins' toolbox is nice. There is a final concern with the idea itself that this will become bloat, more legacy code that never gets updated that the admins need to scroll past, but we can burn that bridge when we come to it.
There is one change that I would suggest, the ban message makes it seem you are banned from observer instead of everything but observer so I would reword it a little.

**Comment by KingofHONK** [comment]
call it a voyeur ban

**Comment by Kurfursten** [comment]
Vetoed by high administration

**Comment by Eneocho** [comment]
rip

**Comment by Securitymain** [comment]
It's over
